lumispy/signals/common_luminescence.py

```diff
@@ -21,47 +21,89 @@
+        region, from out the edges inwards. Cropping can happen uniformily or by specifying the croping range for each axis or each side.
 
         Parameters
         ----------
-        crop_px : int
-            Amount of pixels to be cropped on each side individually.
+        crop_range : int, float, tuple
+            Amount of pixels to be cropped. If an number or a 1-tuple is passed, all sides are cropped by the same amount. If a 2-tuple is passed (crop_x, crop_y), a different amount of pixels is cropped from the x and y directions, respectively. If a 4-tuple is passed (crop_left, crop_bottom, crop_right, crop_top), a different amount of pixels is cropped from each edge individually.
+
+        crop_units : str
+            Select in which units cropping happens. It can be in `pixel` units (default), or in `percent`/`%` units.
 
         Returns
         -------
         signal_cropped : CommonLuminescence
-            A smaller cropped CL signal object. If inplace is True, the original
-            object is modified and no LumiSpectrum is returned.
+            A smaller cropped CL signal object.
         """
 
-        width = self.axes_manager.shape[0]
-        height = self.axes_manager.shape[1]
-
-        if crop_px * 2 > width or crop_px * 2 > height:
+        units_accepted = ("px", "pixel", "percent", "%")
+        if crop_units.lower() not in units_accepted:
             raise ValueError(
-                "The pixels to be cropped cannot be larger than half the width or the length!"
+                "The param crop_units only accepts `pixel` units or `percent`."
             )
+
+        w = self.axes_manager.shape[0]
+        h = self.axes_manager.shape[1]
+        crop_range_type = type(crop_range)
+
+        if crop_range_type in (int, float):
+            crop_vals = [crop_range] * 4
+        elif crop_range_type is tuple:
+            if len(crop_range) == 2:
+                crop_vals = (list(crop_range) * 2,)
+                crop_vals = crop_vals[0]
+            elif len(crop_range) == 4:
+                crop_vals = list(crop_range)
+            else:
+                raise ValueError(
+                    f"The crop_range tuple must be either a 2-tuple (x,y) or a 4-tuple (left, bottom, right, top). You provided a {len(crop_range)}-tuple."
+                )
         else:
-            signal_cropped = self.inav[
-                crop_px + 1 : width - crop_px + 1, crop_px + 1 : height - crop_px + 1
-            ]
+            raise ValueError(
+                f"The crop_range value must be a number or a tuple, not a {crop_range_type}"
+            )
 
-        # Store transformation in metadata (or update the value if already previously transformed)
+        # Negative means reverse indexing
+        crop_vals = array(crop_vals) * [1, -1, -1, 1]
+
+        # Convert percentages to pixel units
+        if crop_units.lower() in units_accepted[-2:]:
+            crop_vals = crop_vals * array([w, h] * 2)
```

Comment:
This assumes that a percentage is given as value between 0 and 1, which is not clear from the docstring (especially as the unit could be `%`)!